pkg/agent/controller/agent_test.go

```diff
@@ -512,7 +514,20 @@ func (c *cluster) init(syncerConfig broker.SyncerConfig) {
 
 func (c *cluster) start(t *testDriver, syncerConfig broker.SyncerConfig) {
 	syncerConfig.LocalClient = c.localDynClient
-	agentController, err := controller.New(&c.agentSpec, syncerConfig, c.localKubeClient)
+	bigint, err := rand.Int(rand.Reader, big.NewInt(1000000))
+	Expect(err).To(Succeed())
+
+	serviceImportCounterName := "submariner_service_import" + bigint.String()
+
+	bigint, err = rand.Int(rand.Reader, big.NewInt(1000000))
+	Expect(err).To(Succeed())
+
+	serviceExportCounterName := "submariner_service_export" + bigint.String()
+
+	agentController, err := controller.New(&c.agentSpec, syncerConfig, c.localKubeClient,
+		controller.AgentConfig{
+			ServiceImportCounterName: serviceImportCounterName,
+			ServiceExportCounterName: serviceExportCounterName})
```

Comment:
Can we verify that the metrics are updated where appropriate?

Reply:
I believe this patch does not bump those metrics, at least I don't see the code

Reply:
They should be updated by the syncer (in Admiral).

Reply:
true but for white box testing we're not supposed to know those details. But knowing those details, it verifies that the admiral components were setup correctly.  Not a big deal either way.